### PR TITLE
Feature change order

### PR DESCRIPTION
**Pull request by IgorOsiyuk**
change user model that can store an array of user's keeps
change POST-request that saves id of new keep in the user collection
change keep's GET-request  that return all keeps of the user from user collection
create new PUT-request that changes the order in user's keeps array

### CODE REVIEW

File: services/keepService.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import { Schema } from 'mongoose';
 import * as serviceTypes from './serviceTypes';
 import KeepModel from '../models/Keep';
+import UserModel, { UserInterface } from '../models/User';
 
 const getKeep = async (userId: Schema.Types.ObjectId) => {
-    const keeps = await KeepModel.find({ userId }).select("-userId -__v")
+    const keeps = await UserModel.find({ _id: userId }).populate("keeps").select('keeps -_id')
```
Comment (melonholy) [inline services/keepService.ts:7]: use findOne

File: services/keepService.ts [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
     if (!keeps) {
         const error = [{
             msg: 'Something goes wrong Our droids fix it tomorrow'
         }]
         throw error
     }
-    return keeps
+    return keeps[0].keeps
```
Comment (melonholy) [inline services/keepService.ts:14]: return `keeps`

**Review by melonholy — CHANGES_REQUESTED** [review]

File: services/keepService.ts (hunk 1 above)
Comment (IgorOsiyuk) [reply services/keepService.ts:7]: - [+]

File: services/keepService.ts (hunk 2 above)
Comment (IgorOsiyuk) [reply services/keepService.ts:14]: - [+]

**Review by melonholy — APPROVED** [review]

**Pull request merged**